pydantic/main.py

```diff
@@ -211,6 +215,19 @@ def __init__(__pydantic_self__, **data: Any) -> None:
     def __get_pydantic_core_schema__(cls, source: type[BaseModel], gen_schema: GenerateSchema) -> CoreSchema:
         return gen_schema.model_schema(cls)
 
+    @classmethod
+    def __pydantic_init_subclass__(cls, **kwargs: Any) -> None:
```

Comment:
I think it's a bit weird to include `kwargs` but omit the keyword arguments that `ModelMetaclass.__new__` takes by name.

I think this should be called with no arguments, but up for debate.

If we do pass `kwargs` please can we add something to the docstring.

Reply:
Well, for what it's worth, I just did it the same way that we handle the call to `super().__new__` (which will call `__init_subclass__` with the same kwargs) in line 109:

https://github.com/pydantic/pydantic/blob/0cd57b3e0c75d9622012de542e5ddc110fe55743/pydantic/main.py#L109

`__init_subclass__` will generally receives the keyword arguments that were passed in the class declaration:

```python
class MyClass:
    def __init_subclass__(cls, **kwargs):
        print(kwargs)

class MySubClass(MyClass, x=1):
    pass

#> {'x': 1}
```
which is why I wanted to pass them to `__pydantic_init_subclass__`.

I'm happy to add the other keyword arguments to the init_subclass call though; should we update the line 109 call too? (I think either way we probably should call with the same keyword arguments that the actual `__init_subclass__` receives.)

Reply:
I'll note that there are tests that explicitly check the behavior of init_subclass with kwargs (and break if you remove the **kwargs from the call to `super().__new__` in the metaclass). So I think at most we may want to add the generic/namespace stuff into the `__init_subclass__` call.

I'm not sure what the right thing to do is (i.e., including the generic stuff and `__pydantic_reset_parent_namespace__`); happy to do it either way.

Reply:
I confirmed we actually can't pass any of those keyword arguments to the standard `__init_subclass__`, so I'm not not inclined to pass them to `__pydantic_init_subclass__` either. Based on the way `__init_subclass__` works my impression is that the right thing to do here is to not pass those things to `__pydantic_init_subclass__` (especially considering you'll basically have access to them anyway since the `cls` will be initialized). If you still disagree I think maybe we should discuss over voice

Reply:
okay, agreed.